### PR TITLE
Introducing 'dependency-tree.yaml', combined 'config.yaml' in App and 'component.yaml' in Components

### PR DESCRIPTION
**Pull request by this**
This PR carries the necessary updates to be comply with the new changes in the UUF Maven plugin (came with [ PR 31](https://github.com/wso2/carbon-uuf-maven-tools/pull/31).

### CODE REVIEW

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/api/config/ComponentManifest.java [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+         * Sets the mode of this binding.
+         *
+         * @param mode mode to be set
+         * @exception IllegalArgumentException if the {@code mode} is not {@link #MODE_PREPEND} or {@link #MODE_APPEND}
+         *                                     or {@link #MODE_OVERWRITE}
+         * @see #MODE_PREPEND
+         * @see #MODE_APPEND
+         * @see #MODE_OVERWRITE
+         */
+        public void setMode(String mode) {
```
Comment (Kishanthan) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/api/config/ComponentManifest.java]: on the config side, let's give only one option. let's go with simple for the config side eg ; append

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/api/config/DependencyNode.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+    /**
+     * Sets the dependencies list of the UUF Component reflected by this node.
+     *
+     * @param dependencies dependencies list to be set
+     */
+    public void setDependencies(List<DependencyNode> dependencies) {
+        if (dependencies == null) {
+            this.dependencies = Collections.emptyList();
+            this.allDependencies = Collections.emptySet();
```
Comment (Kishanthan) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/api/config/DependencyNode.java]: Move these two initialisations to class level or constructer and have the not null check and then the else logic.

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/core/deployment/AppCreator.java [hunk 3]
```diff
@@ -76,49 +83,56 @@ public AppCreator(Set<RenderableCreator> renderableCreators, ClassLoaderProvider
...
 
+            String componentName = dependencyNode.getArtifactId();
+            String componentVersion = dependencyNode.getVersion();
+            String componentContextPath;
+            if (dependencyNode == rootNode) {
+                // This is the root component.
+                componentContextPath = Component.ROOT_COMPONENT_CONTEXT_PATH;
+            } else {
+                componentContextPath = dependencyNode.getContextPath();
+            }
```
Comment (Kishanthan) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/core/deployment/AppCreator.java]: We could merge this into a single line check. Remove the if else branch.

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/io/ArtifactComponentReference.java [hunk 4]
```diff
@@ -101,10 +101,10 @@ private String getExtension(Path filePath) {
     }
 
     @Override
-    public Optional<FileReference> getBindingsConfig() {
-        Path bindingsConfiguration = componentDirectory.resolve(FILE_NAME_BINDINGS);
-        if (Files.exists(bindingsConfiguration)) {
-            return Optional.of(new ArtifactFileReference(bindingsConfiguration, appReference));
+    public Optional<FileReference> getManifest() {
+        Path manifest = componentDirectory.resolve(FILE_NAME_MANIFEST);
```
Comment (Kishanthan) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/io/ArtifactComponentReference.java]: Use componentManifest for the variable name to improve readability.

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/api/config/DependencyNode.java (hunk 2 above)
Comment (this) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/api/config/DependencyNode.java]: fixed in commit 42dfb9bc07ef9747e014a3786516024457dd3852

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/core/deployment/AppCreator.java (hunk 3 above)
Comment (this) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/core/deployment/AppCreator.java]: fixed in commit d5f7fc4b5a83254785aa94a2dc4c726c7778aad8

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/io/ArtifactComponentReference.java (hunk 4 above)
Comment (this) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/io/ArtifactComponentReference.java]: fixed in commit cf95d7ce5007725c2265adf14275055079eff7ed

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/api/config/ComponentManifest.java (hunk 1 above)
Comment (this) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/api/config/ComponentManifest.java]: fixed in commit 2213b6b

**Comment by Kishanthan** [comment]
Fixes #66

**Pull request merged**